### PR TITLE
CeKit config for AMQ Streams Proxy (for 2.7)

### PR DESCRIPTION
**Pull request by k-wall**
This is the CeKit config for the AMQ Stream Proxy that will be part of AMQ Stream 2.7 as a Tech Preview.

### CODE REVIEW

**Comment by ppatierno** [comment]
@k-wall I see it's in Draft but when ready, the PR should be changed to be against a new amqstreams27-dev branch.
It's something that @OwenCorrigan76 @kyguy or I can create when needed.

File: proxy/modules/proxy/module.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+envs:
+  - name: "KROXYLICIOUS_HOME"
+    value: "/opt/kroxylicious"
+  - name: "COM_REDHAT_COMPONENT"
+    value: "amqstreams-proxy-container"
+
+artifacts:
+  - md5: 7dbd6531b76a6b97778a198adfd2dcd1
+    name: kroxylicious-app.zip
+#  - md5: 599d39bb14781bed4bd66eff00e131d1
```
Comment (k-wall) [inline proxy/modules/proxy/module.yaml:15]: TODO: https://issues.redhat.com/browse/ENTMQSTPR-10

**Comment by k-wall** [comment]
There's still the task to add the licenses to the image, but I intend to work on that separately.  Apart from that, I believe everything should be in good shape.

**Review by OwenCorrigan76 — COMMENTED** [review]
`still the task to add the licenses to the image, but I intend to work on that separately`
Is that why lines 13 - 15 are commented out in proxy/install.sh?

**Comment by k-wall** [comment]
> `still the task to add the licenses to the image, but I intend to work on that separately` Is that why lines 13 - 15 are commented out in proxy/install.sh?

Yes, that's part of the same piece of work. Prod Gates requires the licenses to be included in the image.    That's the step to extract the licenses into the image.

File: proxy/modules/proxy/module.yaml (hunk 1 above)
Comment (k-wall) [reply proxy/modules/proxy/module.yaml]: That's fixed.

Comment (k-wall) [reply proxy/modules/proxy/module.yaml]: Task completed.

**Comment by k-wall** [comment]
> still the task to add the licenses to the image, but I intend to work on that separately Is that why lines 13 - 15 are commented out in proxy/install.sh?

That work is now complete.  The licenses are being included in the image.

**Comment by OwenCorrigan76** [comment]
Having discussed MD5 as placeholders with @k-wall, everything else looks in place to me.

File: proxy/modules/proxy/install.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+# untar the artifact containing license information
+mkdir -p ${PRODUCT_LICENSE_DIR}
+mv ${SOURCES_DIR}/kroxylicious-app-licenses.tar.gz ${LICENSE_DIR}
+tar -xvzf ${LICENSE_DIR}/kroxylicious-app-licenses.tar.gz -C ${PRODUCT_LICENSE_DIR} --no-same-owner
+
+# create destination folder of scripts, jars and config
+mkdir -p ${KROXYLICIOUS_HOME}
+
+# unzip archive/artifact with bridge scripts and jars
```
Comment (SamBarker) [inline proxy/modules/proxy/install.sh:20]: nit
```suggestion
# unzip archive/artifact with proxy scripts and jars
```

Comment (SamBarker) [reply proxy/modules/proxy/install.sh:20]: Also why use zip here at all and not tgz?

**Review by SamBarker — APPROVED** [review]
LGTM

A couple of nits to consider.

File: proxy/gating.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+id: amq-streams-release
+product_versions:
+  - amqstreams*
+  - cvp
+decision_contexts:
+  - cvp_default
+rules:
+  # Rules are in format <namespace>.<type>.<category>
+  # TODO - need proxy test suite
+  #- !PassingTestCaseRule {test_case_name: amq-streams.systemtest.x86_64}
```
[the remark this answers is not in the document]
Comment (franvila) [reply proxy/gating.yaml:12]: @im-konge could you help me with that? I'm not aware of what's going on with the gating rules.

File: proxy/modules/proxy/install.sh (hunk 2 above)
Comment (k-wall) [reply proxy/modules/proxy/install.sh]: The remainder of AMQ Streams uses zips for the licence artefact.  Consistency won out.

**Pull request merged**